nncf/common/tensor_statistics/statistics.py

```diff
@@ -26,23 +26,25 @@ class TensorStatistic(ABC):
     TENSOR_STATISTIC_OUTPUT_KEY = "tensor_statistic_output"
 
     @staticmethod
-    def tensor_eq(tensor1: Tensor, tensor2: Tensor, rtol=1e-6) -> bool:
+    def tensor_eq(tensor1: Tensor, tensor2: Tensor, rtol: float = 1e-6) -> Any:
         return fns.allclose(tensor1, tensor2)
 
     @abstractmethod
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> Any:
         pass
 
 
 class MinMaxTensorStatistic(TensorStatistic):
     MIN_STAT = "min_values"
     MAX_STAT = "max_values"
 
-    def __init__(self, min_values, max_values):
+    def __init__(self, min_values: Tensor, max_values: Tensor):
         self.min_values = min_values
         self.max_values = max_values
 
-    def __eq__(self, other: "MinMaxTensorStatistic") -> bool:
+    def __eq__(self, other: object) -> Any:
+        if not isinstance(other, MinMaxTensorStatistic):
+            return NotImplemented
```

Comment:
```suggestion
        if not isinstance(other, MinMaxTensorStatistic):
            return False
```
Please return False here and elsewhere